models/remoteworkitem.go

```diff
@@ -0,0 +1,21 @@
+package models
+
+// RemoteWorkItem represents a work item as it is stored in the database
+type RemoteWorkItem struct {
+	Fields map[string]interface{}	
+}
+
+// The intregration code for a specific remote provider must populate this.
+type WorkItemMap map[string]interface{}
+
+// Map maps RemoteWorkItem to WorkItem
+func (rwi RemoteWorkItem) MapRemote (wiMap WorkItemMap, wiType WorkItemType) WorkItem {
+	workItem := WorkItem{
+		Type: wiType.ID
+	}	
+	
+	for from_key, to_key := range wiMap {
+		workItem.Fields[to_key] = remoteWorkItem.Fields[from_key]
```

Comment:
You're assuming there will only be 1 field named "title" in the whole doc if you flatten it like that. 

A flat v using the whole path "issue.meta.title" would work tho.

I think I would like to Fetch and Store the RemoteWorkItem as-is, but we can sure flatten it was a step after Store and Map.